File: Java Projects/Coffee Machine (Easy)/Stage 4/CoffeeMachine.java
package machine;

import java.util.Scanner;

public class CoffeeMachine {
    public static void main(String[] args) {
        printState();
        System.out.println("Write action (buy, fill, take):");
        String action = scanner.next();
        switch (action) {
            case "buy":
                buy();
                break;
            case "fill":
                fill();
                break;
            case "take":
                take();
                break;
        }
        System.out.println();
        printState();
    }
    
    private static void printState() {
        String[] identifier = {
                "water",
                "milk",
                "coffee beans",
                "disposable cups",
                "money"
        };
        System.out.println("The coffee machine has:");
        for (int i = 0; i < 5; i++) {
            System.out.println(inventory[i] + " of " + identifier[i]);
        }
    }

    private static void take() {
        System.out.println("I give you " + inventory[4]);
        inventory[4] = 0;
    }

    private static void fill() {
        String[] statements = {
                "Write how many ml of water do you want to add:",
                "Write how many ml of milk do you want to add:",
                "Write how many grams of coffee beans do you want to add:",
                "Write how many disposable cups of coffee do you want to add:"
        };
        for (int i = 0; i < 4; i++) {
            System.out.println(statements[i]);
            inventory[i] += scanner.nextInt();
        }
    }

    private static void buy() {
        System.out.println("What do you want to buy? 1 - espresso, 2 - latte, 3 - cappuccino:");
        int choice = scanner.nextInt();
        switch (choice) {
            case 1:
                buyEspresso();
                break;
            case 2:
                buyLatte();
                break;
            case 3:
                buyCappuccino();
                break;
        }
    }

    private static void buyEspresso() {
        inventory[0] -= 250;
        inventory[2] -= 16;
        inventory[3] -= 1;
        inventory[4] += 4;
    }

    private static void buyLatte() {
        inventory[0] -= 350;
        inventory[1] -= 75;
        inventory[2] -= 20;
        inventory[3] -= 1;
        inventory[4] += 7;
    }

    private static void buyCappuccino() {
        inventory[0] -= 200;
        inventory[1] -= 100;
        inventory[2] -= 12;
        inventory[3] -= 1;
        inventory[4] += 6;
    }

    private static Scanner scanner = new Scanner(System.in);
    private static int[] inventory = {400, 540, 120, 9, 550};
}

File: Python Projects/Coffee Machine (Easy)/Stage 4/coffee_machine.py
available_ml_water = 400
available_ml_milk = 540
available_g_beans = 120
available_cups = 9
available_cash = 550


def buy():
    global available_cups
    print("What do you want to buy? 1 - espresso, 2 - latte, 3 - cappuccino: ")
    choice = int(input())
    if choice == 1:
        create_espresso()
    elif choice == 2:
        create_latte()
    elif choice == 3:
        create_cappuccino()
    available_cups -= 1



def create_espresso():
    global available_ml_water
    global available_g_beans
    global available_cash
    available_cash += 4
    available_ml_water -= 250
    available_g_beans -= 16


def create_latte():
    global available_ml_water
    global available_ml_milk
    global available_g_beans
    global available_cash
    available_cash += 7
    available_ml_water -= 350
    available_ml_milk -= 75
    available_g_beans -= 20


def create_cappuccino():
    global available_ml_water
    global available_ml_milk
    global available_g_beans
    global available_cash
    available_cash += 6
    available_ml_water -= 200
    available_ml_milk -= 100
    available_g_beans -= 12


def fill_supplies():
    global available_ml_water
    global available_ml_milk
    global available_g_beans
    global available_cups

    print("Write how many ml of water do you want to add: ")
    available_ml_water += int(input())
    print("Write how many ml of milk do you want to add: ")
    available_ml_milk += int(input())
    print("Write how many grams of coffee beans do you want to add: ")
    available_g_beans += int(input())
    print("Write how many disposable cups of coffee do you want to add: ")
    available_cups += int(input())


def take():
    global available_cash
    temp = available_cash
    available_cash = 0
    print(f"I gave you ${available_cash}")
    return temp


def print_status():
    global available_ml_water
    global available_ml_milk
    global available_g_beans
    global available_cups
    global available_cash
    print("The coffee machine has: ")
    print(f"{available_ml_water} of water")
    print(f"{available_ml_milk} of milk")
    print(f"{available_g_beans} of coffee beans")
    print(f"{available_cups} of disposable cups")
    print(f"{available_cash} of money")


def run():
    print_status()
    while True:
        print()
        print("Write action (buy, fill, take):")
        choice = input()
        if choice == "buy":
            buy()
        elif choice == "fill":
            fill_supplies()
        elif choice == "take":
            take()
        print()
        print_status()
        break


run()
